src/Animated.test.js

```diff
@@ -0,0 +1,177 @@
+  });
+
+  it('fails if decay does not attach nodes correctly', () => {
+    expect(
+      checkIfAttachAndDetachNodesProperly({
+        node: decay,
+        name: 'decay',
+        config: {
+          deceleration: 0.997,
+        },
+      })
+    ).toBeTruthy();
+  });
+
+  it('fails if spring does not attach nodes correctly', () => {
+    expect(
+      checkIfAttachAndDetachNodesProperly({
+        node: spring,
+        name: 'spring',
+        config: {
+          toValue: 0,
+          damping: 7,
+          mass: 1,
+          stiffness: 121.6,
+          overshootClamping: false,
+          restSpeedThreshold: 0.001,
+          restDisplacementThreshold: 0.001,
+        },
+      })
+    ).toBeTruthy();
+  });
+
+  it('fails if animation related nodes are still attached after detaching of value', () => {
+    const { timing, Value } = Animated;
+    class TestComponent extends React.Component {
+      constructor(props) {
+        super(props);
+        this.transX = new Value(0);
+        const config = {
+          duration: 5000,
+          toValue: -120,
+          easing: Easing.inOut(Easing.ease),
+        };
+        this.anim = timing(this.transX, config);
+        this.anim2 = timing(this.transX, config);
+      }
+      start1() {
+        this.anim.start();
+      }
+      start2() {
+        this.anim2.start();
+      }
+      render() {
+        return (
+          <Animated.View style={{ transform: [{ translateX: this.transX }] }} />
+        );
+      }
+    }
+    const ref = React.createRef();
+    const wrapper = renderer.create(<TestComponent ref={ref} />);
+
+    ref.current.start1();
+    ref.current.start2();
+
+    wrapper.unmount();
+    expect(ReanimatedModule.getNumberOfNodes()).toBe(0);
+  });
+
+  it('fails if animation related nodes are detached if there are two children and only one detach', () => {
+    const { timing, Value } = Animated;
+    const transX = new Value(0);
+    const wrapper1 = renderer.create(
+      <Animated.View
+        style={{
+          transform: [{ translateX: transX }],
+        }}
+      />
+    );
+    const wrapper2 = renderer.create(
+      <Animated.View
+        style={{
+          transform: [{ translateX: transX }],
+        }}
+      />
+    );
+    const config = {
+      duration: 5000,
+      toValue: -120,
+      easing: Easing.inOut(Easing.ease),
+    };
+    const anim = timing(transX, config);
+    anim.start();
+    const numberOfNodesBeforeDetach = ReanimatedModule.getNumberOfNodes();
+    wrapper1.unmount();
+    const numberOfNodesAfterDetach = ReanimatedModule.getNumberOfNodes();
+    const result =
+      // 3 means AnimatedProps, AnimatedStyle and AnimatedTransform
+      // which are nodes not related to animation and has to be detached
+      numberOfNodesBeforeDetach - 3 === numberOfNodesAfterDetach &&
+      numberOfNodesAfterDetach > 3;
+    wrapper2.unmount();
+    expect(result && ReanimatedModule.getNumberOfNodes() === 0).toBeTruthy();
```

Comment:
we should check if result is true immediately, then do another check that nodes are at 0 after second unmount. This way if the test fails it is straightforward to tell which of these two conditions has failed. You can have as many expect statements in a test as you wish